### PR TITLE
Terrain Generation

### PR DESCRIPTION
**Pull request by naomijub**
addresses https://github.com/rewin123/space_editor/issues/94 

- [x] terrain heightmap generation
- [x] meshed terrain 
- [ ] Biomes
- [ ] cubic/quadratic terrain grid
- [ ] hexagonal terrain grid (hexx crate)
- [ ] feature toggle terrain render system (Mesh, quadratic grid, hexagonal grid, just heightmap - for voxels)
- [ ] feature for terrain physics/collision
- [ ] UNIT TESTS!
- [ ] examples
- [ ] Improve noise function 
- [ ] Have this crate as a portable standalone crate for other projects
- [ ] Persistence 
- [ ] Select vertex
- [ ] Edit specific vertex information

### CODE REVIEW

**Comment by rewin123** [comment]
I suggest bringing this first before merging it into 0.4. I may be join to work on pr tommorrow

**Comment by naomijub** [comment]
> I suggest this for first merging into 0.4. I may be join to work on pr tommorrow

My plans for the PR:
- [ ] feature toggle terrain render system (Mesh, quadratic grid, hexagonal grid, just heightmap - for voxels)
- [ ] feature for terrain physics/collision
- [x] meshed terrain 
- [ ] cubic/quadratic terrain grid
- [ ] hexagonal terrain grid (hexx crate)
- [ ] UNIT TESTS!
- [ ] examples
- [ ] Improve noise function

File: modules/terrain_gen/src/mesh/mod.rs [hunk 1]
```diff
@@ -120,7 +118,13 @@ impl Generation for TerrainMesh {
...
             let face_normal = v1.cross(v2).normalize();
 
-            normals.push(face_normal);
+            normals[i_0] += face_normal;
+            normals[i_1] += face_normal;
+            normals[i_2] += face_normal;
+        }
+
+        for normal in normals.iter_mut() {
+            *normal = normal.normalize();
```
Comment (naomijub) [inline modules/terrain_gen/src/mesh/mod.rs:127]: Definitive forgot this part hahaha

**Comment by naomijub** [comment]
addresses https://github.com/rewin123/space_editor/issues/94

**Comment by naomijub** [comment]
I will come back to this PR on Jan 3

**Comment by naomijub** [comment]
I messed up this branch, sorry

**Comment by naomijub** [comment]
Closing this in favour of a future brand new PR

**Pull request closed without merging**